From 7b80328845bda9c17424170dc9366ca97ca4c7f9 Mon Sep 17 00:00:00 2001
From: David Barbion <davidb@230ruedubac.fr>
Date: Tue, 27 Aug 2024 12:49:43 +0200
Subject: [PATCH] add rpi5_64 platform

---
 Makefile | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Makefile b/Makefile
index a3c2c85..f8501eb 100644
--- a/Makefile
+++ b/Makefile
@@ -275,6 +275,18 @@ else ifeq ($(platform), rpi4_64)
    SINGLE_PRECISION_FLOATS = 1
    MIN_AUDIO_PROCESSING_PER_FRAME = 1
    HAVE_NEON = 1
+else ifeq ($(platform), rpi5_64)
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   SHARED := -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T -Wl,-no-undefined
+   CFLAGS += -DARM -DGNU_SOURCE=1
+   CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a76
+   CFLAGS += -fomit-frame-pointer -fstrict-aliasing
+   CFLAGS += -fno-exceptions
+   CXXFLAGS += -fno-rtti
+   SINGLE_PRECISION_FLOATS = 1
+   MIN_AUDIO_PROCESSING_PER_FRAME = 1
+   HAVE_NEON = 1
 else ifeq ($(platform), xenon)
    TARGET := $(TARGET_NAME)_libretro_xenon360.a
    CC = xenon-gcc$(EXE_EXT)
-- 
2.46.0

