--- a/source/frontends/libretro/libretro.cpp	2025-01-26 22:25:07.000000000 +0100
+++ b/source/frontends/libretro/libretro.cpp	2025-01-27 12:52:36.203625847 +0100
@@ -391,7 +391,7 @@
     try
     {
         const size_t size = ra2::RetroSerialisation::getSize();
-        ra2::log_cb(RETRO_LOG_INFO, "RA2: %s - size = %" SIZE_T_FMT "\n", __FUNCTION__, size);
+        //ra2::log_cb(RETRO_LOG_INFO, "RA2: %s - size = %" SIZE_T_FMT "\n", __FUNCTION__, size);
         return size;
     }
     catch (const std::exception &e)
