diff --git a/pcsx2/GameDatabase.cpp b/pcsx2/GameDatabase.cpp
index b3a2814..69d30f8 100644
--- a/pcsx2/GameDatabase.cpp
+++ b/pcsx2/GameDatabase.cpp
@@ -103,7 +103,7 @@ GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::entryFromYaml(const std::str
 			bool fixValidated = false;
 			for (GamefixId id = GamefixId_FIRST; id < pxEnumEnd; id++)
 			{
-				std::string validFix = fmt::format("{}Hack", wxString(EnumToString(id)).ToUTF8());
+				std::string validFix = fmt::format("{}Hack", wxString(EnumToString(id)).ToStdString());
 				if (validFix == fix)
 				{
 					fixValidated = true;
@@ -129,7 +129,7 @@ GameDatabaseSchema::GameEntry YamlGameDatabaseImpl::entryFromYaml(const std::str
 				bool speedHackValidated = false;
 				for (SpeedhackId id = SpeedhackId_FIRST; id < pxEnumEnd; id++)
 				{
-					std::string validSpeedHack = fmt::format("{}SpeedHack", wxString(EnumToString(id)).ToUTF8());
+					std::string validSpeedHack = fmt::format("{}SpeedHack", wxString(EnumToString(id)).ToStdString());
 					if (validSpeedHack == speedHack)
 					{
 						speedHackValidated = true;
diff --git a/pcsx2/Recording/InputRecording.cpp b/pcsx2/Recording/InputRecording.cpp
index 18ed6a6..8c31679 100644
--- a/pcsx2/Recording/InputRecording.cpp
+++ b/pcsx2/Recording/InputRecording.cpp
@@ -286,7 +286,7 @@ void InputRecording::SetupInitialState(u32 newStartingFrame)
 	if (state != InputRecordingMode::Replaying)
 	{
 		inputRec::log("Started new input recording");
-		inputRec::consoleLog(fmt::format("Filename {}", inputRecordingData.GetFilename().ToUTF8()));
+		inputRec::consoleLog(fmt::format("Filename {}", inputRecordingData.GetFilename().ToStdString()));
 		SetToRecordMode();
 	}
 	else
@@ -298,7 +298,7 @@ void InputRecording::SetupInitialState(u32 newStartingFrame)
 
 		incrementUndo = true;
 		inputRec::log("Replaying input recording");
-		inputRec::consoleMultiLog({fmt::format("File: {}", inputRecordingData.GetFilename().ToUTF8()),
+		inputRec::consoleMultiLog({fmt::format("File: {}", inputRecordingData.GetFilename().ToStdString()),
 								   fmt::format("PCSX2 Version Used: {}", std::string(inputRecordingData.GetHeader().emu)),
 								   fmt::format("Recording File Version: {}", inputRecordingData.GetHeader().version),
 								   fmt::format("Associated Game Name or ISO Filename: {}", std::string(inputRecordingData.GetHeader().gameName)),
@@ -317,7 +317,7 @@ void InputRecording::SetupInitialState(u32 newStartingFrame)
 
 void InputRecording::FailedSavestate()
 {
-	inputRec::consoleLog(fmt::format("{} is not compatible with this version of PCSX2", savestate.ToUTF8()));
+	inputRec::consoleLog(fmt::format("{} is not compatible with this version of PCSX2", savestate.ToStdString()));
 	inputRec::consoleLog(fmt::format("Original PCSX2 version used: {}", inputRecordingData.GetHeader().emu));
 	inputRecordingData.Close();
 	initialLoad = false;
@@ -388,14 +388,14 @@ bool InputRecording::Play(wxWindow* parent, wxString filename)
 										 L"Savestate files (*.p2s)|*.p2s", wxFD_OPEN);
 			if (loadStateDialog.ShowModal() == wxID_CANCEL)
 			{
-				inputRec::consoleLog(fmt::format("Could not locate savestate file at location - {}", savestate.ToUTF8()));
+				inputRec::consoleLog(fmt::format("Could not locate savestate file at location - {}", savestate.ToStdString()));
 				inputRec::log("Savestate load failed");
 				inputRecordingData.Close();
 				return false;
 			}
 
 			savestate = loadStateDialog.GetPath();
-			inputRec::consoleLog(fmt::format("Base savestate set to {}", savestate.ToUTF8()));
+			inputRec::consoleLog(fmt::format("Base savestate set to {}", savestate.ToStdString()));
 		}
 		state = InputRecordingMode::Replaying;
 		initialLoad = true;
@@ -421,7 +421,7 @@ void InputRecording::GoToFirstFrame(wxWindow* parent)
 			if (!initiallyPaused)
 				g_InputRecordingControls.PauseImmediately();
 
-			inputRec::consoleLog(fmt::format("Could not locate savestate file at location - {}\n", savestate.ToUTF8()));
+			inputRec::consoleLog(fmt::format("Could not locate savestate file at location - {}\n", savestate.ToStdString()));
 			wxFileDialog loadStateDialog(parent, _("Select a savestate to accompany the recording with"), L"", L"",
 										 L"Savestate files (*.p2s)|*.p2s", wxFD_OPEN);
 			int result = loadStateDialog.ShowModal();
@@ -434,7 +434,7 @@ void InputRecording::GoToFirstFrame(wxWindow* parent)
 				return;
 			}
 			savestate = loadStateDialog.GetPath();
-			inputRec::consoleLog(fmt::format ("Base savestate swapped to {}", savestate.ToUTF8()));
+			inputRec::consoleLog(fmt::format ("Base savestate swapped to {}", savestate.ToStdString()));
 		}
 		StateCopy_LoadFromFile(savestate);
 	}
@@ -454,7 +454,7 @@ wxString InputRecording::resolveGameName()
 	{
 		if (IGameDatabase* gameDB = AppHost_GetGameDatabase())
 		{
-			GameDatabaseSchema::GameEntry game = gameDB->findGame(std::string(gameKey.ToUTF8()));
+			GameDatabaseSchema::GameEntry game = gameDB->findGame(std::string(gameKey.ToStdString()));
 			if (game.isValid)
 			{
 				gameName = fromUTF8(game.name);
diff --git a/pcsx2/gui/AppCoreThread.cpp b/pcsx2/gui/AppCoreThread.cpp
index 1512ba5..9c2b767 100644
--- a/pcsx2/gui/AppCoreThread.cpp
+++ b/pcsx2/gui/AppCoreThread.cpp
@@ -314,7 +314,7 @@ static int loadGameSettings(Pcsx2Config& dest, const GameDatabaseSchema::GameEnt
 	// TODO - config - this could be simplified with maps instead of bitfields and enums
 	for (SpeedhackId id = SpeedhackId_FIRST; id < pxEnumEnd; id++)
 	{
-		std::string key = fmt::format("{}SpeedHack", wxString(EnumToString(id)).ToUTF8());
+		std::string key = fmt::format("{}SpeedHack", wxString(EnumToString(id)).ToStdString());
 
 		// Gamefixes are already guaranteed to be valid, any invalid ones are dropped
 		if (game.speedHacks.count(key) == 1)
@@ -331,7 +331,7 @@ static int loadGameSettings(Pcsx2Config& dest, const GameDatabaseSchema::GameEnt
 	// TODO - config - this could be simplified with maps instead of bitfields and enums
 	for (GamefixId id = GamefixId_FIRST; id < pxEnumEnd; id++)
 	{
-		std::string key = fmt::format("{}Hack", wxString(EnumToString(id)).ToUTF8());
+		std::string key = fmt::format("{}Hack", wxString(EnumToString(id)).ToStdString());
 
 		// Gamefixes are already guaranteed to be valid, any invalid ones are dropped
 		if (std::find(game.gameFixes.begin(), game.gameFixes.end(), key) != game.gameFixes.end())
