diff --git a/package/llvm-project/llvm-project.mk b/package/llvm-project/llvm-project.mk
index 5859ff7cfc..6f075793a5 100644
--- a/package/llvm-project/llvm-project.mk
+++ b/package/llvm-project/llvm-project.mk
@@ -5,7 +5,7 @@
 ################################################################################
 
 LLVM_PROJECT_VERSION_MAJOR = 15
-LLVM_PROJECT_VERSION = $(LLVM_PROJECT_VERSION_MAJOR).0.3
+LLVM_PROJECT_VERSION = $(LLVM_PROJECT_VERSION_MAJOR).0.7
 LLVM_PROJECT_SITE = https://github.com/llvm/llvm-project/releases/download/llvmorg-$(LLVM_PROJECT_VERSION)
 
 include $(sort $(wildcard package/llvm-project/*/*.mk))
