From 9d1d985b6b0d346a4fb21799847c7849a362f5f8 Mon Sep 17 00:00:00 2001
From: David Barbion <davidb@230ruedubac.fr>
Date: Tue, 11 Mar 2025 09:05:32 +0100
Subject: [PATCH 1/9] steamdeck enable gamepad on startup

---
 drivers/hid/hid-steam.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/hid/hid-steam.c b/drivers/hid/hid-steam.c
index 10460b7bde1a..aa8965251a2e 100644
--- a/drivers/hid/hid-steam.c
+++ b/drivers/hid/hid-steam.c
@@ -1247,6 +1247,7 @@ static int steam_probe(struct hid_device *hdev,
 	spin_lock_init(&steam->lock);
 	mutex_init(&steam->report_mutex);
 	steam->quirks = id->driver_data;
+	steam->gamepad_mode = true; // enable gamepad at startup
 	INIT_WORK(&steam->work_connect, steam_work_connect_cb);
 	INIT_DELAYED_WORK(&steam->mode_switch, steam_mode_switch_cb);
 	INIT_LIST_HEAD(&steam->list);
-- 
2.52.0

