From fbd514b32268fc5b4df8df2278eb9017d3e9d819 Mon Sep 17 00:00:00 2001
From: Pit64 <yrigaud@icloud.com>
Date: Wed, 19 Mar 2025 10:31:29 +0100
Subject: [PATCH] recalbox paths

---
 src/platform/sdl2/main.cpp | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/src/platform/sdl2/main.cpp b/src/platform/sdl2/main.cpp
index 6e719ab..f26f4fe 100644
--- a/src/platform/sdl2/main.cpp
+++ b/src/platform/sdl2/main.cpp
@@ -532,7 +532,7 @@ int main(int argc, char **argv) {
     }
 
     int w, h;
-    SDL_GameControllerAddMappingsFromFile("gamecontrollerdb.txt");
+    SDL_GameControllerAddMappingsFromFile("/recalbox/share/system/configs/openlara/gamecontrollerdb.txt");
     SDL_Init(SDL_INIT_VIDEO|SDL_INIT_AUDIO|SDL_INIT_EVENTS|SDL_INIT_GAMECONTROLLER|SDL_INIT_HAPTIC);
 
     SDL_GetCurrentDisplayMode(0, &sdl_displaymode);
@@ -565,16 +565,12 @@ int main(int argc, char **argv) {
 
     SDL_ShowCursor(SDL_DISABLE);
 
-    const char *home;
-    if (!(home = getenv("HOME")))
-        home = getpwuid(getuid())->pw_dir;
-    strcat(cacheDir, home);
-    strcat(cacheDir, "/.openlara/");
+    strcat(cacheDir, "/recalbox/share/system/configs/openlara/");
 
     struct stat st = {0};
     if (stat(cacheDir, &st) == -1 && mkdir(cacheDir, 0777) == -1)
         cacheDir[0] = 0;
-    strcpy(saveDir, cacheDir);
+    strcat(saveDir, "/recalbox/share/saves/tombraider/");
 
     timeval t;
     gettimeofday(&t, NULL);
-- 
2.49.0

