diff --git a/package/wlroots/wlroots.mk b/package/wlroots/wlroots.mk
index fdd2fb8250..485f92f698 100644
--- a/package/wlroots/wlroots.mk
+++ b/package/wlroots/wlroots.mk
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-WLROOTS_VERSION = 0.16.2
+WLROOTS_VERSION = 0.17.1
 WLROOTS_SITE = https://gitlab.freedesktop.org/wlroots/wlroots/-/releases/$(WLROOTS_VERSION)/downloads
 WLROOTS_LICENSE = MIT
 WLROOTS_LICENSE_FILES = LICENSE
@@ -22,7 +22,8 @@ WLROOTS_DEPENDENCIES = \
 	seatd \
 	udev \
 	wayland \
-	wayland-protocols
+	wayland-protocols \
+	libdisplay-info
 
 WLROOTS_CONF_OPTS = -Dexamples=false -Dxcb-errors=disabled
 
