diff --git a/common/command.c b/common/command.c
index e5d9b9c..9193746 100644
--- a/common/command.c
+++ b/common/command.c
@@ -503,7 +503,7 @@ static int cmd_call(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 	return result;
 }
 
-enum command_ret_t cmd_process(int flag, int argc, char * const argv[],
+int cmd_process(int flag, int argc, char * const argv[],
 			       int *repeatable, ulong *ticks)
 {
 	enum command_ret_t rc = CMD_RET_SUCCESS;
diff --git a/drivers/power/charge_animation.c b/drivers/power/charge_animation.c
index 5ed7bd8..df78f67 100644
--- a/drivers/power/charge_animation.c
+++ b/drivers/power/charge_animation.c
@@ -545,7 +545,7 @@ static int charge_animation_show(struct udevice *dev)
 	ulong delta;
 	ulong ms = 0, sec = 0;
 	int start_idx = 0, show_idx = -1, old_show_idx = IMAGE_RESET_IDX;
-	int soc, voltage, current, key_state;
+	int soc = 0, voltage = 0, current = 0, key_state = 0;
 	int i, charging = 1, ret;
 #ifdef CONFIG_RKIMG_BOOTLOADER
 	int boot_mode;
