diff --git a/package/initscripts/init.d/rcS b/package/initscripts/init.d/rcS
index de411534da..e95f91b7e8 100755
--- a/package/initscripts/init.d/rcS
+++ b/package/initscripts/init.d/rcS
@@ -1,5 +1,8 @@
 #!/bin/sh
 
+while read -r _ENV; do
+  export "${_ENV?}"
+done < /etc/environment
 
 # Start all init scripts in /etc/init.d
 # executing them in numerical order.
@@ -20,6 +23,7 @@ for i in /etc/init.d/S??* ;do
 	    ;;
 	*)
 	    # No sh extension, so fork subprocess.
+	    recallog -s "rcS" -t "INIT" -f init.log "start $i"
 	    $i start
 	    ;;
     esac
