From 6e902fea2a240d694aaf73ea2b66633e29da1819 Mon Sep 17 00:00:00 2001
From: Bkg2k <bkg2k9@gmail.com>
Date: Wed, 23 Feb 2022 16:14:53 +0100
Subject: [PATCH 06/28] use passwords in configuration file

---
 network/netplay/netplay_frontend.c | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/network/netplay/netplay_frontend.c b/network/netplay/netplay_frontend.c
index 52dc297ada..0235457ea5 100644
--- a/network/netplay/netplay_frontend.c
+++ b/network/netplay/netplay_frontend.c
@@ -847,8 +847,8 @@ static void handshake_password(void *userdata, const char *line)
    if (netplay_send(&connection->send_packet_buffer, connection->fd, &password_buf, sizeof(password_buf)))
       netplay_send_flush(&connection->send_packet_buffer, connection->fd, false);
 
-   menu_input_dialog_end();
-   retroarch_menu_running_finished(false);
+   //menu_input_dialog_end();
+   //retroarch_menu_running_finished(false);
 }
 #endif
 
@@ -1121,7 +1121,7 @@ bool netplay_handshake_init(netplay_t *netplay,
    if (!netplay->is_server)
    {
       /* If a password is demanded, ask for it */
-      if ((connection->salt = ntohl(header[3])))
+      /*if ((connection->salt = ntohl(header[3])))
       {
 #ifdef HAVE_MENU
          menu_input_ctx_line_t line = {0};
@@ -1132,7 +1132,7 @@ bool netplay_handshake_init(netplay_t *netplay,
          if (!menu_input_dialog_start(&line))
             return false;
 #endif
-      }
+      }*/
 
       if (!netplay_handshake_nick(netplay, connection))
          return false;
@@ -1143,6 +1143,16 @@ bool netplay_handshake_init(netplay_t *netplay,
    *had_input       = true;
    netplay_recv_flush(&connection->recv_packet_buffer);
 
+   /* If a password is demanded, ask for it */
+   if (!netplay->is_server && (connection->salt = ntohl(header[3])))
+   {
+      settings_t *settings = config_get_ptr();
+      const char* password = (settings->bools.netplay_start_as_spectator) ?
+                              settings->paths.netplay_spectate_password : settings->paths.netplay_password;
+      handshake_password(NULL, password);
+      netplay_recv_flush(&connection->recv_packet_buffer);
+   }
+
    return true;
 
 error:
-- 
2.51.0

