From 055acfb9d3bf27beb7955051e523e523c9d78ec6 Mon Sep 17 00:00:00 2001
From: digitalLumberjack <digitallumberjack@gmail.com>
Date: Sat, 22 Jun 2024 22:21:44 +0200
Subject: [PATCH 14/28] crt: add support for several resolution for switchres

---
 gfx/video_crt_switch.c | 18 +++++++++++++++---
 1 file changed, 15 insertions(+), 3 deletions(-)

diff --git a/gfx/video_crt_switch.c b/gfx/video_crt_switch.c
index 0797e4cd05..30b2a21519 100644
--- a/gfx/video_crt_switch.c
+++ b/gfx/video_crt_switch.c
@@ -188,14 +188,26 @@ static bool crt_sr2_init(videocrt_switch_t *p_switch,
             RARCH_LOG("[CRT]: CRT Mode: %d - arcade_15 \n", crt_mode);
             break;
          case 2:
+            sr_set_monitor("arcade_15_25");
+            RARCH_LOG("[CRT]: CRT Mode: %d - arcade_15_25 \n", crt_mode);
+            break;
+         case 3:
             sr_set_monitor("arcade_31");
             RARCH_LOG("[CRT]: CRT Mode: %d - arcade_31 \n", crt_mode);
             break;
-         case 3:
+         case 4:
+            sr_set_monitor("arcade_15_31");
+            RARCH_LOG("[CRT]: CRT Mode: %d - arcade_15_31 \n", crt_mode);
+            break;
+         case 5:
+            sr_set_monitor("arcade_15_25_31");
+            RARCH_LOG("[CRT]: CRT Mode: %d - arcade_15_25_31 \n", crt_mode);
+          break;
+         case 6:
             sr_set_monitor("pc_31_120");
             RARCH_LOG("[CRT]: CRT Mode: %d - pc_31_120 \n", crt_mode);
-            break;
-         case 4:
+          break;
+         case 7:
             RARCH_LOG("[CRT]: CRT Mode: %d - Selected from ini \n", crt_mode);
             break;
          default:
-- 
2.51.0

