From edf7461fd33e4ed08fb89adb2cb225e4cb2ebedf Mon Sep 17 00:00:00 2001
From: digitalLumberjack <digitallumberjack@gmail.com>
Date: Tue, 8 Jul 2025 12:17:26 +0200
Subject: [PATCH 19/28] crt: adjust system rates before reset context fix
 dolphin on crt

---
 retroarch.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/retroarch.c b/retroarch.c
index 9ec6c8dbad..32d5c006ed 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -1585,7 +1585,10 @@ void drivers_init(
 
       if (   !(video_st->flags & VIDEO_FLAG_CACHE_CONTEXT_ACK)
             && hwr->context_reset)
+      {
+         driver_adjust_system_rates(runloop_st, video_st, settings);
          hwr->context_reset();
+      }
       video_st->flags            &= ~VIDEO_FLAG_CACHE_CONTEXT_ACK;
       runloop_st->frame_time_last = 0;
    }
-- 
2.51.0

