From 16ba86ce01af60aa1d338f661e72fddd2645b1be Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Mon, 11 Mar 2024 18:36:51 +0000
Subject: [PATCH 174/176] ffconf: Add loop option for race testing

---
 pi-util/ffconf.py | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/pi-util/ffconf.py b/pi-util/ffconf.py
index 71cd838720..7024612006 100755
--- a/pi-util/ffconf.py
+++ b/pi-util/ffconf.py
@@ -198,6 +198,8 @@ def doconf(csva, tests, test_root, vcodec, dectype, args):
     else:
         print("All tests normal:", successes, "ok,", failures, "failed")
 
+    return unx_failures + unx_success
+
 
 class ConfCSVDialect(csv.Dialect):
     delimiter = ','
@@ -222,6 +224,7 @@ if __name__ == '__main__':
     argp.add_argument("--ffmpeg", default="./ffmpeg", help="ffmpeg exec name; if directory given use <dir>/ffmpeg")
     argp.add_argument("--valgrind", action='store_true', help="Run valgrind on tests")
     argp.add_argument("--gen_yuv", action='store_true', help="Create yuv file (stored with log under /tmp)")
+    argp.add_argument("--loop", default=0, type=int, help="Create yuv file (stored with log under /tmp)")
     args = argp.parse_args()
 
     if not os.path.isdir(args.test_root):
@@ -254,5 +257,11 @@ if __name__ == '__main__':
         print("FFmpeg file '%s' not found" % args.ffmpeg)
         exit(1)
 
-    doconf(csva, args.tests, args.test_root, args.vcodec, dectype, args)
+    i = 0
+    while True:
+        i = i + 1
+        if args.loop:
+            print("== Loop ", i)
+        if doconf(csva, args.tests, args.test_root, args.vcodec, dectype, args) or (args.loop >= 0 and i > args.loop):
+            break
 
-- 
2.46.0

