From b180a5b79963c71f1d7dedde40dffb96bbeb35e2 Mon Sep 17 00:00:00 2001
From: Bkg2k <bkg2k9@gmail.com>
Date: Mon, 23 Aug 2021 15:33:35 +0200
Subject: [PATCH 7/9] fix race condition in make depencency chain

---
 Makefile.linux | 1 +
 rules.mak      | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile.linux b/Makefile.linux
index 531e2aa..b98ccc4 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -90,6 +90,7 @@ ti99sim:
 	    fi \
 	  fi; \
 	  for dir in $(SRC_DIRS); do \
+	    mkdir -p $$dir/$(CFG); \
 	    $(MAKE) -sC $$dir $(TI_DATA_DIR); \
 	  done \
 	fi
diff --git a/rules.mak b/rules.mak
index 4dc2dff..ca4d283 100644
--- a/rules.mak
+++ b/rules.mak
@@ -122,4 +122,4 @@ $(CFG)/%.o : %.m
 all:
 
 $(CFG):
-	@mkdir $(CFG)
+	@mkdir -p $(CFG)
-- 
2.46.0

